requirements.txt

```diff
@@ -14,3 +14,4 @@ jsonfield~=3.1.0
 django-private-storage~=2.2
 celery~=4.4.0
 django-ipware~=3.0.0
+djangosaml2~=1.2.0
```

Comment:
Let's make this dependency optional (eg: https://github.com/openwisp/openwisp-utils/blob/8331e52d32eea77056f9c600c1ff770dafbc8a19/setup.py#L48-L61). I'm ok with getting rid of `requirements.txt` and move the entire definition to `setup.py`, which is simpler (the method used now is just something that was copied over and over from the past).

Reply:
`requirements.txt` makes specifying dependencies from GitHub easier or cleaner in my opinion. I would like it stay.

Reply:
What do you mean specifying dependencies from Github? Editing via the web interface?
It doesn't sound like a good reason to keep some unnecessary code around to me, moreover, the only way to specify this as optional dependency compatible with `setup.py` is to use `extra_requires`, let's simplify please. I hate this `requirements.txt` parsing code repeated in every module, it's really dumb and it was my mistake to put it there in the first place but it's never too late to fix it.

Reply:
Ok, got it.